src/darwin/Framework/CHIP/MTROTAImageTransferHandler.mm

```diff
@@ -28,9 +28,16 @@
 
 constexpr uint32_t kMaxBdxBlockSize = 1024;
 
+constexpr double kMilliSecondsInSecond = 1000.0;
+
 // Timeout for the BDX transfer session. The OTA Spec mandates this should be >= 5 minutes.
 constexpr System::Clock::Timeout kBdxTimeout = System::Clock::Seconds16(5 * 60);
 
+// For thread devices, we need to throttle sending Blocks in response to BlockQuery messages
+// to avoid spamming the network with too many BDX messages. We are going to match the polling
+// interval of 50 ms as the time to wait before sending a Block in response to a BlockQuery.
```

Comment:
```suggestion
// For thread devices, we may want to throttle sending Blocks in response to BlockQuery messages
// to avoid spamming the network with too many BDX messages.  For now, match the old 50ms
// polling interval we used to have.
```

But this should really (followup) become a configurable quantity.

Reply:
woody already asked for it so i am going to do it with this PR